### PR TITLE
feat: TS typings for request module

### PR DESCRIPTION
**Pull request by eoln**
TypeScript typings for `request` module

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -568,6 +569,39 @@ declare namespace SDKStandardComponents {
...
+    interface RequestResponse {
+        statusCode: number
+        headers?: Record<string, string>
+        data: string | Buffer | Record<string, unknown>
+    }
+
+    function request(opts: RequestOptions): Promise<RequestResponse>
+
+    namespace request {
+        var ResponseType: RequestResponseType
```
Comment (lewisdaly) [inline src/index.d.ts:603]: Should this be `let` instead of `var`?

File: src/index.d.ts [hunk 2]
```diff
@@ -568,6 +569,39 @@ declare namespace SDKStandardComponents {
...
+        Text = Symbol('text'),
+        Stream = Symbol('stream')
+    }
+    enum RequestMethod {
+        GET = 'GET',
+        PATCH = 'PATCH',
+        POST = 'POST',
+        PUT = 'PUT'
+    }
+    interface RequestOptions {
```
Comment (kleyow) [inline src/index.d.ts:582]: This is based on the options found here, right? https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/request.js#L23

Shouldn't qs/querystring be apart of this interface?

Comment (kleyow) [reply src/index.d.ts:582]: As well as `responseType`.

**Comment by eoln** [comment]
@kleyow @lewisdaly : PR comments addressed and fixed.

File: src/index.d.ts (hunk 2 above)
Comment (eoln) [reply src/index.d.ts:576]: thanks @kleyow fixed

File: src/index.d.ts (hunk 1 above)
Comment (eoln) [reply src/index.d.ts]: fixed

**Review by kleyow — APPROVED** [review]

**Review by lewisdaly — APPROVED** [review]
Looks good to me - do you also need to bump the `package.json` version?

**Pull request merged**